platform/core/src/services/HangingProtocolService/HangingProtocolService.js

```diff
@@ -54,6 +66,24 @@ class HangingProtocolService {
   addProtocols(protocols) {
     protocols.forEach(protocol => {
       if (this.protocols.indexOf(protocol) === -1) {
+        protocol.id = protocol.id || protocol.name;
+        // Automatically compute some number of attributes if they
+        // aren't present.  Makes defining new HPs easier.
+        protocol.name = protocol.name || protocol.id;
+        const { stages } = protocol;
+
+        // Generate viewports automatically as required.
+        stages.forEach(stage => {
```

Comment:
This is just helper code to make it easier to define hanging protocols - it could all be done manually